### PR TITLE
Overlay virtual room call events into main timeline

### PR DESCRIPTION
**Pull request by kerryarchibald**
Fixes https://github.com/vector-im/element-web/issues/22929

[Virtual rooms documentation](https://gitlab.matrix.org/new-vector/voip/matrix-sip-bridge/-/blob/master/docs/virtual-room-mapping.md) (only internal for now, docs in element-meta are in progress with product team)

Limitations:
- Doesn't handle proper backpaging
- Doesn't handle unread or message counts for virtual events in main room counts

## Checklist

* [ ] Tests written for new code (and old code if feasible)
* [ ] Linter and other CI checks pass
* [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Overlay virtual room call events into main timeline ([\#9626](https://github.com/matrix-org/matrix-react-sdk/pull/9626)). Fixes vector-im/element-web#22929. Contributed by @kerryarchibald.

### CODE REVIEW

**Comment by Morhonda** [comment]
Hi @kerryarchibald,

could you give me an update regarding the call history in virtual rooms for Element Web? Thanks. We have a call with the client tomorrow morning.

Best wishes,

Moritz

**Review by robintown — COMMENTED** [review]
Looks good, I only have minor nits

**Review by robintown — APPROVED** [review]
I still want to know if something can be done with the `roomView.setProps({})` hack, but otherwise this looks good

File: test/components/structures/RoomView-test.tsx [hunk 1]
```diff
@@ -108,6 +111,7 @@ describe("RoomView", () => {
             </SDKContext.Provider>,
         );
         await act(() => Promise.resolve()); // Allow state to settle
+        roomView.setProps({});
```
[the remark this answers is not in the document]
Comment (kerryarchibald) [reply test/components/structures/RoomView-test.tsx]: Using rtl seems to solve the issue with the UI not updating

**Pull request merged**